### PR TITLE
Allow selecting tickets to place order and redirect to orders page - Part 1

### PR DESCRIPTION
**Pull request by pradeepgangwar**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Currently user cannot select tickets and place a order. This PR allows user to select tickets for which he wants to place order and redirect them to orders page (Orders page is not dynamic now. That will be done after this PR is merged)

#### Changes proposed in this pull request:
- Add new actions to select tickets and save order.
- Modify models where ever necessary.

Part of #750

### CODE REVIEW

**Comment by CosmicCoder96** [comment]
hey @pradeepgangwar I am not sure if you guys have already discussed this, but this is the flow of how the orders are supposed to be placed according to a discussion I had with @niranjan94 a while ago. See if this is of any help 😄 

> - User is in public event page /e/{id}
> - User picks tickets
> - We temporarily generate a UUID on the client side and store the tickets selected, user, event references to localstorage.
> - We load /orders/{client-side-uuid}/new and we pick data up from localstorage and use.
> - User fills in holder details etc etc.
> - User presses buy/order. We make server request, (If server request succeeded, we delete localstorage record and start using the server record/uuid for all further steps).

**Comment by pradeepgangwar** [comment]
@CosmicCoder96 There is slight change in approach we are using, although most of the part is same except that we are not using local storage and interacting with server at every moment. :smile: We had a little discussion with @niranjan94 in gitter channel and we discussed this approach with him. Here is full implementation how we plan to approach this:

1. User selects the tickets for the order.
2. We create order/attendees on the FE. Push the attendees(with random names) so that tickets are locked. Push the order on the server with the relationship with attendees and event to get the hash. (I'll remove the requirement constraints from other fields so that we can create the order).
3. You can then update the names to blank on the FE(don't push to server)
4. User get's the form, and 10 minutes slot. If he/she fills it, great. We update the order with the status "placed" and other details and start the payment procedure using the charges api. Also update the attendee names. After the payment succeeds we update the status to "completed" and send tickets/notifications otherwise we set the status to "expired". We send back the error. Show the payment error page with the info about the error(v1 does it).
5. If they weren't able to enter the info, simply expire the order and show them an alert that the order has expired and redirect to the public page(v1 does it) or anywhere you want.

Feel free to suggest any changes here :smile:

**Comment by srv-twry** [comment]
@CosmicCoder96 

> We temporarily generate a UUID on the client side and store the tickets selected, user, event references to localstorage.

I just have one problem with the approach that you're suggesting. We need to push the order to the server so that the tickets are locked on the server.

**Comment by srv-twry** [comment]
@CosmicCoder96 
I like the approach that you're suggesting but after a little modification.

1. User is in public event page /e/{id}
2. User picks tickets and we create attendees on the server with random names(lock the tickets).
3. We temporarily generate a UUID on the client side and store the tickets selected, user, event references to localstorage.
4. We load /orders/{client-side-uuid}/new and we pick data up from localstorage and use.
5. User fills in holder details etc etc.
6. User presses buy/order. We make server request, (If server request succeeded, we delete localstorage record and start using the server record/uuid for all further steps). We update the holder details on the server now.
7. If the details weren't filled in 10 mins or payment wasn't successful, delete the attendees created on the server so that the tickets are unlocked now. Also update the order status to expired.

Both the ways discussed works for me. I have completely implemented the server side in accordance with what @pradeepgangwar has stated above so if possible it would be better to stick to it if it doesn't have loop holes in it.

**Comment by srv-twry** [comment]
@ritikamotwani we don't have any other alternatives. We have to create dummy attendees for both the ideas.

**Comment by ritikamotwani** [comment]
Oh okay @srv-twry, got it!

File: app/controllers/public/index.js [hunk 1]
```diff
@@ -11,6 +11,34 @@ export default Controller.extend({
...
+          await attendee.save();
+        }
+        order.set('attendees', attendees.slice());
+        await order.save()
+          .then(order => {
+            this.get('notify').success(this.get('l10n').t('Order details saved. Please fill further details within 10 minutes.'));
+            this.transitionToRoute('orders.new', order.identifier);
+          })
+          .catch(e => {
+            console.error(e);
```
Comment (ritikamotwani) [inline app/controllers/public/index.js]: @pradeepgangwar this line that gives a console log can be removed.

File: app/models/ticket.js [hunk 2]
```diff
@@ -30,8 +30,8 @@ export default ModelBase.extend({
    * Relationships
    */
   event           : belongsTo('event'),
-  order           : belongsTo('order'),
-  attendee        : belongsTo('attendee'),
+  order           : hasMany('order'),
+  attendee        : hasMany('attendee'),
```
Comment (srv-twry) [inline app/models/ticket.js]: `orders` and `attendees`. Plural :)

**Comment by pradeepgangwar** [comment]
@ritikamotwani @srv-twry Did the changes. Please review again.

File: app/components/public/ticket-list.js [hunk 3]
```diff
@@ -25,6 +25,36 @@ export default Component.extend(FormMixin, {
...
+        }
+      }
+    },
+    placeOrder() {
+      let order = this.get('order');
+      let event = order.get('event');
+      order.tickets.forEach(ticket => {
+        let attendee = ticket.orderQuantity;
+        let i;
+        for (i = 0; i < attendee; i++) {
```
Comment (srv-twry) [inline app/components/public/ticket-list.js]: Do we have some sort of `forEach` loop in JavaScript like we have in Java ?  This `let i` doesn't look good.

Comment (srv-twry) [reply app/components/public/ticket-list.js]: Sorry, this is the quantity. Please rename attendee to numberOfAttendees or equivalent.

Comment (pradeepgangwar) [reply app/components/public/ticket-list.js]: @srv-twry `forEach` is used to iterate over array. We don't have array here.

**Pull request merged**